### PR TITLE
feat: Adjust license validation methods

### PR DESCRIPTION
**Pull request by matthewBearCamunda**
## Description

Implement requested PR changes from https://github.com/camunda/camunda/pull/21156

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

closes #https://github.com/camunda/camunda/pull/21156

### CODE REVIEW

File: service/src/main/java/io/camunda/service/license/CamundaLicense.java [hunk 1]
```diff
@@ -38,8 +40,13 @@ public synchronized String getLicenseType() {
 
   public synchronized void initializeWithLicense(final String license) {
     if (!isInitialized) {
-      isValid = determineLicenseValidity(license);
-      licenseType = getLicenseTypeFromProperty(license);
+      if (StringUtils.isNotBlank(license)) {
```
Comment (nicpuppa) [inline service/src/main/java/io/camunda/service/license/CamundaLicense.java:43]: 🔧 I'm not a big fan of `StringUtils.isNotBlank(license)`, I'd suggest to use a native approach: `license != null && !license.isBlank()`. To me it's a bit more clear, and we can get rid off the import 😄 

🔧 We can also refactor a bit this method removing some if branching

```
public synchronized void initializeWithLicense(final String license) {
    if (isInitialized) {
        return;
    }

    if (license != null && !license.isBlank()) {
        isValid = determineLicenseValidity(license);
        licenseType = getLicenseTypeFromProperty(license);
    } else {
        LOGGER.error("No license detected when one is expected. Please provide a license.");
    }

    isInitialized = true;
}
```

**Review by nicpuppa — APPROVED** [review]
Thanks @matthewBearCamunda

File: service/src/main/java/io/camunda/service/license/CamundaLicense.java [hunk 2]
```diff
@@ -37,12 +38,18 @@ public synchronized String getLicenseType() {
...
+      return;
+    }
+
+    if (license != null && !license.isBlank()) {
       isValid = determineLicenseValidity(license);
       licenseType = getLicenseTypeFromProperty(license);
-
-      isInitialized = true;
+    } else {
+      LOGGER.error("No license detected when one is expected. Please provide a license.");
```
Comment (nicpuppa) [inline service/src/main/java/io/camunda/service/license/CamundaLicense.java:49]: ❓ If the license is not provided we logged an error. But what happens next ? It's not clear to me

Comment (matthewBearCamunda) [reply service/src/main/java/io/camunda/service/license/CamundaLicense.java:49]: I added an action the customer is expected to take to fix the error in the message

But as to what will happen, the license validity will be marked as false, and the licenseType will be set to `unknown`. I added explicit setters to this block to make that clear

**Pull request merged**